src/components/admin/manage/media/MediaDetailModal.tsx

```diff
@@ -0,0 +1,414 @@
+import { useState } from "react";
+import { toast } from "sonner";
+import {
+	AlertDialog,
+	AlertDialogAction,
+	AlertDialogCancel,
+	AlertDialogContent,
+	AlertDialogDescription,
+	AlertDialogFooter,
+	AlertDialogHeader,
+	AlertDialogTitle,
+} from "@/components/ui/alert-dialog";
+import { Button } from "@/components/ui/button";
+import {
+	Dialog,
+	DialogContent,
+	DialogHeader,
+	DialogTitle,
+} from "@/components/ui/dialog";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Textarea } from "@/components/ui/textarea";
+import { useDeleteMedia } from "@/feature/media/queries/useDeleteMedia";
+import type { MediaFile } from "@/feature/media/types/media";
+import { formatDate } from "@/lib/utils";
+
+type Props = {
+	media: MediaFile | null;
+	isOpen: boolean;
+	onClose: () => void;
+	onUpdate?: (media: MediaFile) => void;
+};
+
+export default function MediaDetailModal({
+	media,
+	isOpen,
+	onClose,
+	onUpdate,
+}: Props) {
+	const t = useTranslations("Media");
+	const tCommon = useTranslations("Common");
+	const deleteMediaMutation = useDeleteMedia();
+	// const [isEditing, setIsEditing] = useState(false);
+	const [showDeleteConfirm, setShowDeleteConfirm] = useState(false);
+	const [metadata, setMetadata] = useState({
+		altText: "",
+		caption: "",
+		description: "",
+		title: "",
+	});
+	const locale = useLocale();
+
+	if (!media) return null;
+
+	const formatFileSize = (bytes: number) => {
+		if (bytes === 0) return "0 Bytes";
+		const k = 1024;
+		const sizes = ["Bytes", "KB", "MB", "GB"];
+		const i = Math.floor(Math.log(bytes) / Math.log(k));
+		return parseFloat((bytes / Math.pow(k, i)).toFixed(2)) + " " + sizes[i];
+	};
+
+	const copyToClipboard = async (text: string) => {
+		try {
+			await navigator.clipboard.writeText(text);
+			toast.success(tCommon("CopiedToClipboard"));
+		} catch (error) {
+			console.error("Failed to copy text: ", error);
+			toast.error(tCommon("ErrorCopiedToClipboard"));
+		}
+	};
+
+	const handleDownload = () => {
+		const link = document.createElement("a");
+		link.href = media.url;
+		link.download = media.filename;
+		document.body.appendChild(link);
+		link.click();
+		document.body.removeChild(link);
+	};
+
+	const handleDelete = () => {
+		setShowDeleteConfirm(true);
+	};
+
+	const confirmDelete = async () => {
+		try {
+			await deleteMediaMutation.mutateAsync(media.filename);
+			toast.success(t("DeleteSuccess"));
+			setShowDeleteConfirm(false);
+			onClose();
+		} catch (error) {
+			console.error("Error deleting media:", error);
+			toast.error(t("DeleteError"));
+			setShowDeleteConfirm(false);
+		}
+	};
+
+	const cancelDelete = () => {
+		setShowDeleteConfirm(false);
+	};
+
+	const renderMediaPreview = () => {
+		if (media.category === "image") {
+			return (
+				<div className="relative w-full h-full overflow-hidden">
+					<Image
+						alt={media.filename}
+						className="object-cover"
```

Comment:
```suggestion
						className="object-contain"
```

to avoid zoomed media and get a full display of the image

Reply:
In Figma, the image is zoomed.
If you want the original and full image you have the link "View Media File"